### ISSUE TITLE
Removed lines starting with 2 dashes are breaking diff output

### ISSUE THREAD

**Issue by emohamed** [rtfpessoa/diff2html#87]
Diff2html fails to recognise a line as removed when it starts with 2 or more dashes:

```
--- broken
+++ broken
@@ -1,15 +1,12 @@
 test
 
----
+test 
 
 test  
----
\ No newline at end of file
```

Here is proof of the problem: http://diff2html-bug-preview.surge.sh/

**Comment by rtfpessoa** [comment]
@emohamed I think you found a bug. I will look into this.

**Comment by rtfpessoa** [comment]
Just released version `diff2html@2.0.0-rc.7`, it should fix this issue.
Please test and give me some feedback if something is still missing.

**Comment by emohamed** [comment]
Works like a charm! 

Thanks for the quick fix :+1:

**Comment by emohamed** [comment]
Actually, this case is still broken: 
https://gist.github.com/emohamed/dc5c9449dec39a0fdbacaf2f2ca535f1

**Comment by rtfpessoa** [comment]
I was actually thinking about it today and I remembered that case. (which is kind of obvious, but I missed it)

But actually I cannot find an easy solution I think I have to do some minor refactor to better find the beginning of files in the diff.

I will let you know when I have something.

**Comment by rtfpessoa** [comment]
@emohamed I just released `2.0.0-rc.8` now I added more through tests. Let me know if you find any other cases.
